File: database/src/main/java/com/usharik/database/WordInfo.java
package com.usharik.database;

public record WordInfo(
    Long wordId,
    String word,
    String[][] cases,
    String translation_ru,
    String translation_en,
    String gender,
    String declensionType
) {
    public WordInfo(Long wordId, String word, String[][] cases, String translation_ru, String translation_en, String gender, String declensionType) {
        this.wordId = wordId;
        this.word = word == null ? "" : word;
        this.cases = cases;
        this.translation_ru = translation_ru == null ? "" : translation_ru;
        this.translation_en = translation_en == null ? "" : translation_en;
        this.gender = gender == null ? "" : gender;
        this.declensionType = declensionType == null ? "" : declensionType;
    }

    public String cases(int i, int j) {
        return this.cases[i][j];
    }
}

File: database/src/main/kotlin/com/usharik/database/WordInfo.kt
package com.usharik.database


/**
 * Immutable dictionary entry shared by the data layer and both quiz modes.
 *
 * The component-style accessors preserve Java-call compatibility while the
 * rest of the app is migrated incrementally to Kotlin properties.
 */
data class WordInfo(
    val wordId: Long?,
    val word: String?,
    val cases: Array<Array<String>>?,
    val translation_ru: String?,
    val translation_en: String?,
    val gender: String?,
    val declensionType: String?,
) {
    fun wordId() = wordId
    fun word() = word.orEmpty()
    fun cases() = cases
    fun translation_ru() = translation_ru.orEmpty()
    fun translation_en() = translation_en.orEmpty()
    fun gender() = gender.orEmpty()
    fun declensionType() = declensionType.orEmpty()
    fun cases(number: Int, grammaticalCase: Int) = cases!![number][grammaticalCase]
}
